src/dune_rules/merlin/merlin.ml

```diff
@@ -351,7 +351,11 @@ module Unprocessed = struct
       Path.Set.singleton
       @@ obj_dir_of_lib `Private mode (Obj_dir.of_local obj_dir)
     in
-    let flags = Ocaml_flags.common flags in
+    let flags =
+      match mode with
+      | `Ocaml -> Ocaml_flags.common flags
+      | `Melange -> Ocaml_flags.get flags Melange
+    in
```

Comment:
Do you know why `Ocaml_flags.get` isn't used also for the OCaml mode? This is actually the only usage of `Ocaml_flags.common` that I could find in the whole repo. When building modules we use `get`:

https://github.com/ocaml/dune/blob/ece416166f6696fa565dc2f36594ec2e0ecd0f85/src/dune_rules/module_compilation.ml#L205

Maybe this can be replaced with

```suggestion
    let flags = Ocaml_flags.get flags mode
```

The only change is that merlin would take into account specific flags for OCaml compilation, but I only see that as an improvement? 🤔 

If you decide to do that, we could remove `common` from `Ocaml_flags.mli`.

Reply:
Maybe. One thing I noticed is that `common` doesn't have `-g`, which you can actually see in the test output.

Reply:
This use of `common` does seem weird. We can leave it alone for now and save it for another PR

Reply:
I don't know if there's anything actionable here. I'm assuming "leave it alone for now" means "keep common" since that's how it was in the first place.

Reply:
The PR looks fine for me. I'm waiting for @jchavarri to approve